tenseal/cpp/tensors/ckksvector.cpp

```diff
@@ -510,27 +509,52 @@ shared_ptr<CKKSVector> CKKSVector::enc_matmul_plain_inplace(
 }
 
 shared_ptr<CKKSVector> CKKSVector::replicate_first_slot_inplace(size_t n) {
-    if (this->_ciphertexts.size() != 1)
-        throw invalid_argument(
-            "can't execute replicate_first_slot on chunked vectors");
-
+    auto slot_count = this->tenseal_context()->slot_count<CKKSEncoder>();
     // mask
-    vector<double> mask(this->size(), 0);
+    vector<double> mask(min(slot_count, n), 0);
     mask[0] = 1;
-    this->mul_plain_inplace(mask);
+    this->_mul_plain_inplace(this->_ciphertexts[0], mask);
+    Ciphertext masked = this->_ciphertexts[0];
 
-    // replicate
-    Ciphertext tmp = this->_ciphertexts[0];
     auto galois_keys = this->tenseal_context()->galois_keys();
-    for (size_t i = 0; i < (size_t)ceil(log2(n)); i++) {
-        this->tenseal_context()->evaluator->rotate_vector_inplace(
-            tmp, static_cast<int>(-pow(2, i)), *galois_keys);
-        this->tenseal_context()->evaluator->add_inplace(this->_ciphertexts[0],
-                                                        tmp);
-        tmp = this->_ciphertexts[0];
+
+    auto replicator = [&](Ciphertext ct, size_t n_repl) -> Ciphertext {
```

Comment:
do we need a copy of `ct` here? instead of ref.

Reply:
Yeah, we will be creating an additional ciphertext if we didn't.